File: iSeeCars.Business/Managers/ModelManager.cs
using iSeeCars.Core.Entities;

namespace iSeeCars.Business.Managers
{
    public class ModelManager
    {
        List<Model> models = new List<Model>();
        private int nextId = 1;

        // Console üçün
        public Model AddModel()
        {
        mdlName:
            Console.WriteLine("Enter a model name:");
            string mdlName = Console.ReadLine();
            if (string.IsNullOrWhiteSpace(mdlName))
            {
                Console.WriteLine("Model name can not be empty");
                goto mdlName;
            }

            Model newModel = new Model
            {
                ModelId = nextId++,
                ModelName = mdlName
            };
            models.Add(newModel);
            return newModel;
        }

        // API üçün
        public Model AddModel(Model model)
        {
            if (model == null || string.IsNullOrWhiteSpace(model.ModelName))
                throw new ArgumentException("Model is null or name is empty");

            model.ModelId = nextId++;
            models.Add(model);
            return model;
        }
        public Model GetModelById(int id)
        {
            var model = models.FirstOrDefault(m => m.ModelId == id);
            if (model == null)
            {
                Console.WriteLine("Model not found");
            }


            return model;

        }
        public List<Model> GetAllModels()
        {
            return models;
        }
        public void ShowAllModels()
        {
            if (models.Count == 0)
            {
                Console.WriteLine("Model not found");
            }
            else
            {
                foreach (var mdl in models)
                {
                    Console.WriteLine($"{mdl.ModelId}.{mdl.ModelName}");
                }

            }
        }
      
        public void RemoveModel(Model model)
        {
            models.Remove(model);
        }
    }
}

File: iSeeCars.java/src/main/java/iseecars/manager/ModelManager.java
package iseecars.manager;

import iseecars.model.Model;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class ModelManager {
    private List<Model> models = new ArrayList<Model>();
    private int nextId = 1;
    private Scanner scanner = new Scanner(System.in);



    // Console Version
    public Model addModel(){
        String mdlName;

        while(true){
            System.out.println("Enter a model name");
            mdlName = scanner.nextLine();

            if(mdlName != null && !mdlName.isBlank()){
                break;
            }
            System.out.println("Model name cannot be empty");
        }
        Model newModel = new Model();
        newModel.setModelId(nextId++);
        newModel.setModelName(mdlName);

        models.add(newModel);
        return newModel;

    }


    // API Version
    public Model addModel(Model model) {
        if (model == null || model.getModelName() == null || model.getModelName().isBlank()) {
            throw new IllegalArgumentException("Model is null or name is empty");
        }

        model.setModelId(nextId++);
        models.add(model);
        return model;
    }

    public Model getModelById(int id){
        return models.stream()
                .filter(m -> m.getModelId() == id)
                .findFirst()
                .orElse(null);
    }
    public List<Model> getAllModels(){
        return models;
    }

    public void showAllModels(){
        if(models.isEmpty()){
            System.out.println("Model not found");
            return;
        }

        for(Model mdl : models){
            System.out.println(mdl.getModelId() + ". " + mdl.getModelName());
        }
    }
    public void removeModel(Model model){
        models.remove(model);
    }


}
